tests/test_model.py

```diff
@@ -0,0 +1,236 @@
+        setup_ran = True
+
+    rw.run(debug=True, model_options={ 'initialization_array': ['one', 'two', 'three'] })
+    assert setup_ran == True
+
+def test_model_options_missing():
+
+    rw = RunwayModel()
+    @rw.setup(options={'initialization_array': array(item_type=text)})
+    def setup(opts):
+        pass
+
+    # this will print to stderr still, but the test should pass
+    with pytest.raises(SystemExit):
+        with pytest.raises(MissingOptionError):
+            rw.run(debug=True)
+
+def test_meta(capsys):
+
+    rw = RunwayModel()
+
+    @rw.setup(options={'initialization_array': array(item_type=text)})
+    def setup(opts):
+        pass
+
+    kwargs_1 = {
+        'inputs': {
+            'image': image,
+            'vector': vector(length=32)
+        },
+        'outputs': {
+            'label': text
+        }
+    }
+    @rw.command('command_1', **kwargs_1)
+    def command_1(opts):
+        pass
+
+    kwargs_2 = {
+        'inputs': {
+            'any': any_type,
+            'file': file
+        },
+        'outputs': {
+            'number': number(min=10, max=100)
+        }
+    }
+
+    @rw.command('command_2', **kwargs_2)
+    def command_2(opts):
+        pass
+
+    expected_manifest = {
+        "options": [{
+            "name": "initialization_array",
+            "type": "array",
+            "itemType": {
+                "type": "text",
+                "name": "text",
+                "default": "",
+                "minLength": 0
+            },
+            "minLength": 0
+        }],
+        "commands": [{
+            "name": "command_1",
+            "inputs": [{
+                "type": "image",
+                "name": "image",
+                "channels": 3
+            }, {
+                "type": "vector",
+                "name": "vector",
+                "length": 32,
+                "samplingMean": 0,
+                "samplingStd": 1
+            }],
+            "outputs": [{
+                "type": "text",
+                "name": "label",
+                "default": "",
+                "minLength": 0
+            }]
+        },
+        {
+            "name": "command_2",
+            "inputs": [{
+                "type": "any",
+                "name": "any"
+                }, {
+                "type": "file",
+                "name": "file"
+            }],
+            "outputs": [{
+                "type": "number",
+                "name": "number",
+                "default": 0,
+                "min": 10,
+                "max": 100,
+                "step": 1
+            }]
+        }]
+    }
+
+    meta_before = os.environ.get('RW_META')
+    os.environ['RW_META'] = '1'
+
+    rw.run(debug=True, model_options={ 'initialization_array': ['one', 'two', 'three'] })
+    std = capsys.readouterr()
+    assert json.loads(std.out) == expected_manifest
+    assert std.err == ''
+
+    if meta_before is not None: os.environ['RW_META'] = meta_before
+
```

Comment:
Might be worth adding two more tests for detecting `SetupError` and `InferenceError`.

Reply:
100%! There are quite a few more model tests to add actually, but I figured some is better than none. I'll add these as well.